src/web/modules/discrepancies/discrepancies.http.ts

```diff
@@ -0,0 +1,35 @@
+import { Request, Response, NextFunction } from 'express'
+
+import logger from './../../../lib/logger'
+import { Connector } from '../../../lib/pay-request/typed_clients/client'
+import { wrapAsyncErrorHandlers } from './../../../lib/routes'
+
+const searchTransaction = async function searchTransaction(req: Request, res: Response) {
+  res.render('discrepancies/report', { csrf: req.csrfToken() })
+}
+
+const getDiscrepancyReport = async function getDiscrepancyReport(req: Request, res: Response) {
+  const chargeIds = req.body.search_string.split(',').map((chargeId: string) => chargeId.trim())
+  const comparisons = await Connector.charges.getGatewayComparisons(chargeIds)
+
+  const interpretedComparisons = comparisons.map((comparison) => {
+    const interpretedComparison = {
+      ...comparison,
+      canBeResolved: comparison.gatewayStatus !== comparison.payExternalStatus
+    }
+    return interpretedComparison
+  })
+
+  res.render('discrepancies/search_results', { comparisons: interpretedComparisons, csrf: req.csrfToken() })
+}
+
+const resolveDiscrepancy = async function resolveDiscrepancy(req: Request, res: Response) {
+  const chargeId = req.params.id
+  const [resolution] = await Connector.charges.resolveDiscrepancy(chargeId)
```

Comment:
Just wondering why this needs to be in square brackets?

Reply:
This will have been copied over from the existing class. The request to resolve discrepancies returns a list of gateway comparisons. As we only try to resolve one discrepancy at a time from toolbox, we're just trying to get the only result from the returned array.